### ISSUE TITLE
Error message for address_pool_gap  

### ISSUE THREAD

**Issue by piotr-iohk** [cardano-foundation/cardano-wallet#204]
| Release | Operating System       | Cause |
| ---     | ---                    | ---   |
| next    | Linux | Code |

# Context
 
While doing `POST v2/wallets` one needs to provide `address_pool_gap` which needs to be between 10 and 100.
When providing the value that is outside range there is an error message thrown `Error in $['address_pool_gap']: ErrGapOutOfRange`. This message should be made more informative and human friendly.

# Steps to Reproduce

1. start cardano wallet launcher:
```
export NETWORK=testnet
stack exec -- cardano-wallet-launcher
```

2.
```
curl  -vX POST http://localhost:8090/v2/wallets \
  -H "Content-Type: application/json; charset=utf-8" \
  -d '{
"name": "Piotr Wallet",
"mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
"passphrase": "Secure Passphrase",
"address_pool_gap": address_pool_gap
}' --http1.1
```
**case 1**
address_pool_gap = -1

**case 2**
address_pool_gap = 20.5

**case 3**
address_pool_gap = 101

## Expected behavior

**case 1, case 2, case 3**
```
Error in $['address_pool_gap']: Provided value is outside of range [10..100]
```
## Actual behavior
 
**case 1**
```
Error in $['address_pool_gap']: Word8 is either floating or will cause over or underflow: -1.0
```

**case 2**
```
Error in $['address_pool_gap']: Word8 is either floating or will cause over or underflow: 20.5
```

**case 3**
```
Error in $['address_pool_gap']: ErrGapOutOfRange 101
```

---

# Resolution Plan

 - It should be enough to modify the 'FromJSON' instance from the Api types to make it use the 'FromText' instance which already has a nice error message.

# PR

| Number   | Base      |
| ---      | ---       |
| #?       | `develop` |

# QA 

- Tests updated in [Cardano.Wallet.Api.TypesSpec](https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs#L234-L244) to illustrate the new message.

- Also, did some manual verifications with _cURL_

**Scenario:** bigger than the maximum
```
 $ curl  -vX POST http://localhost:8090/v2/wallets   -H "Content-Type: application/json; charset=utf-8"   -d '{
"name": "Piotr Wallet",
"mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
"passphrase": "Secure Passphrase",
"address_pool_gap": 250             
}' --http1.1

< HTTP/1.1 400 Bad Request
Error in $['address_pool_gap']: An address pool gap must be a natural number between 10 and 100.
```

**Scenario:** smaller than the minimum
```
$ curl  -vX POST http://localhost:8090/v2/wallets   -H "Content-Type: application/json; charset=utf-8"   -d '{
"name": "Piotr Wallet",
"mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
"passphrase": "Secure Passphrase",
"address_pool_gap": -5 
}' --http1.1
< HTTP/1.1 400 Bad Request
Error in $['address_pool_gap']: An address pool gap must be a natural number between 10 and 100.
```

**Scenario: not a number**
```
$ curl  -vX POST http://localhost:8090/v2/wallets   -H "Content-Type: application/json; charset=utf-8"   -d '{
"name": "Piotr Wallet",
"mnemonic_sentence": ["identify", "screen", "lock", "bargain", "inch", "drop", "canyon", "flock", "dry", "zone", "wash", "argue", "system", "glory", "light"],
"passphrase": "Secure Passphrase",
"address_pool_gap": "patate"
}' --http1.1

< HTTP/1.1 400 Bad Request
Error in $['address_pool_gap']: expected Int, encountered String
```

**Comment by piotr-iohk** [comment]
I need to return that back to `In progress` as still there are few cases producing not expected behaviour:

Some particular negative values (but not all of them), e.g.:
address_pool_gap = -1000 -> creates wallet with `address_pool_gap = 24`
address_pool_gap = -132323000 > creates wallet with `address_pool_gap = 72`


floating point numbers:
address_pool_gap = 2.5 -> `Int is either floating or will cause over or underflow: 2.5` (I'd expect `An address pool gap must be a natural number between 10 and 100.`)

**Comment by KtorZ** [comment]
@piotr-iohk, parser for non-integer & out-of-range values fixed it #221. Putting this back in QA (cf integration tests related to address_pool_gap).

**Comment by piotr-iohk** [comment]
lgtm :+1:
